kms/yubikey/yubikey_test.go

```diff
@@ -988,3 +1017,64 @@ func TestYubiKey_Close(t *testing.T) {
 		})
 	}
 }
+
+func Test_getSerialNumber(t *testing.T) {
+	ok := newStubPivKey(t, RSA)
+
+	failAttest := newStubPivKey(t, RSA)
+	delete(failAttest.attestMap, piv.SlotAuthentication)
+
+	failParse := newStubPivKey(t, ECDSA)
+	serialNumer, err := asn1.Marshal("112233")
+	if err != nil {
+		t.Fatal(err)
+	}
+	attCertParse, err := failParse.attestCA.Sign(&x509.Certificate{
```

Comment:
Why isn't this going to parse successfully?

Reply:
I am adding the serial number extension with a string (ASN.1 PrintableString Type `asn1.Marshal("112233")` while the extension expects an integer. If you look at the function `getSerialNumber()`, we try to unmarshal to an integer, and that test case fails there.